### ISSUE TITLE
Shell Tab broken with dependency injection

### ISSUE THREAD

**Issue by 1357310795** [dotnet/maui#7895]
### Description

I'm using Shell Tab to navigate and dependency injection in my MainPage. But the rendering is broken.
![Screenshot_2022-06-09-21-33-10-618_com companyname mauitabbroken](https://user-images.githubusercontent.com/35869804/172860329-607cf913-e569-4c7c-b9cd-b12503e43f3f.jpg)

### Steps to Reproduce

Minimal Reproducing Example：[MauiTabBroken.zip](https://github.com/dotnet/maui/files/8870932/MauiTabBroken.zip)

1. Create a new MAUI app. (The default template)
2. Add a simple tab in AppShell.xaml
```
<?xml version="1.0" encoding="UTF-8" ?>
<Shell
    x:Class="MauiTabBroken.AppShell"
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:local="clr-namespace:MauiTabBroken">
    
    <Tab Title="Domestic">
        <ShellContent Title="Cats" ContentTemplate="{DataTemplate local:MainPage}" />
        <ShellContent Title="Dogs" ContentTemplate="{DataTemplate local:MainPage}" />
    </Tab>
</Shell>

```
3. Create a new blank ViewModel
```
namespace MauiTabBroken.ViewModel;

public partial class MainViewModel
{
}

```
4. Edit the constructor of MainPage
```
public MainPage(MainViewModel viewModel)
{
	InitializeComponent();
	this.BindingContext = viewModel;
}
```
5. Configurate dependency injection in MauiProgram.cs
```
builder.Services.AddSingleton<MainPage>();
builder.Services.AddSingleton<MainViewModel>();
```
6. Run the program on Android
7. Switch to "Dogs" tab and switch back, you'll see the rendering error.

### Version with bug

6.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android11.0 - API 30

### Did you find any workaround?

When I'm not using the dependency injection, Tab works fine.

**Comment by kristinx0211** [comment]
verified repro on android 12 using above repro project.
![image](https://user-images.githubusercontent.com/99457000/172969720-1c73010b-a257-4ca8-adc2-1dae55f79eb0.png)

**Comment by 1357310795** [comment]
Not fixed on the latest available public version（6.0.486）

**Comment by 1357310795** [comment]
？？？How to reopen？？？
